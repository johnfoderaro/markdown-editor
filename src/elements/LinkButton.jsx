import React from 'react';
import { hot } from 'react-hot-loader';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledButton = styled.button`
  background-color: transparent;
  border: none;
  cursor: pointer;
  font: inherit;
  text-decoration: underline;
  display: inline;
  margin: 0;
  padding: 0;
  &:hover,
  &:focus {
    text-decoration: none;
  }
`;

const LinkButton = ({ children, ...props }) => (
  <StyledButton {...props}>
    {children}
  </StyledButton>
);

LinkButton.propTypes = { children: PropTypes.node.isRequired };

export default hot(module)(LinkButton);
