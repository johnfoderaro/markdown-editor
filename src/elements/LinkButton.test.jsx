import React from 'react';
import { shallow } from 'enzyme';

import LinkButton from './LinkButton';

let tree;

beforeAll(() => {
  tree = shallow(
    <LinkButton>
      Test
    </LinkButton>,
  );
});

describe('<LinkButton />', () => {
  it('should be defined', () => {
    expect(tree).toBeDefined();
  });
  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
