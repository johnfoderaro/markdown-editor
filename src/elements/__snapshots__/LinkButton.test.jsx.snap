// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> should render correctly 1`] = `
<styled.button>
  Test
</styled.button>
`;
